Create c-tag-find/c-today/c-tomorrow feature

yanbingtao opened the pull request:
1. `c-today`:
- Able to list down all persons whose tag(s) match today's day (i.e. Monday, Tuesday, etc).
2. `c-tomorrow`:
- Able to list down all persons whose tag(s) match the next day's day (i.e. Monday, Tuesday, etc).
3. `c-tag-find KEYWORD [KEYWORDS] ...`
- Able to list down all persons whose tag(s) match the keyword(s) entered by the user.

yanbingtao committed:
add c-today feature.

yanbingtao committed:
Move predicates to Model.java.

yanbingtao committed:
Create c-tomorrow feature.

yanbingtao committed:
Create c-tag-find feature.

yanbingtao committed:
Improve code quality by rearranging imports.

yanbingtao committed:
Add parttime/fulltime tags in contact sample data.

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TagFindCommand.java:
```diff
@@ -0,0 +1,42 @@
+import seedu.address.model.person.TagContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
```
Incorrect javadoc? I'm reviewing based on the assumption that this command will find an employee by the tag.

yanbingtao replied:
Indeed, I forgot to change the javadoc after editing from the original "c-find" code. Thanks! I've edited this part.

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TagFindCommand.java:
```diff
@@ -0,0 +1,42 @@
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class TagFindCommand extends Command {
```
I think `FindTagCommand`or `FindByTagCommand`  would be more apt/ sounds better.

yanbingtao replied:
Thank you for the suggestion. Yes, I think that sounds clearer. I've changed it to `FindByTagCommand`. Thanks!

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TagFindCommand.java:
```diff
@@ -0,0 +1,42 @@
+
+    public static final String COMMAND_WORD = "c-tag-find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
```
Same problem as the javadoc

yanbingtao replied:
I've changed it in the latest commit, thanks!

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TodayCommand.java:
```diff
@@ -0,0 +1,57 @@
+
+import seedu.address.model.Model;
+
+public class TodayCommand extends Command {
```
So this is a different feature from the one listed in this PR?

ureshiiYing replied:
It's also missing a javadoc description

yanbingtao replied:
`c-tag-find` was implemented on top of  `c-today` feature, so `c-tag-find` branch contains the `c-today` feature.

yanbingtao replied:
I've added the javadoc. Thanks!

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TodayCommand.java:
```diff
@@ -0,0 +1,57 @@
+            break;
+        default:
+            model.updateFilteredPersonList(person -> false);
+
```
Remove extra line.

yanbingtao replied:
Removed, thanks!

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TodayCommand.java:
```diff
@@ -0,0 +1,57 @@
+            model.updateFilteredPersonList(person -> false);
+
+        }
+        if (model.getFilteredPersonList().size() == 0) {
```
I think it is better to add a blank line above this if-else block

yanbingtao replied:
Added a new line. Thanks!

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TomorrowCommand.java:
```diff
@@ -0,0 +1,58 @@
+        }
+        return new CommandResult(String.format(MESSAGE_SUCCESS, model.getFilteredPersonList().size(),
+                tmrDate.getDayOfWeek()));
+    }
```
Same comments as TodayCommand

yanbingtao replied:
Added new lines in between in the latest commit. Thank you!

ureshiiYing commented on src/main/java/seedu/address/logic/commands/TomorrowCommand.java:
```diff
@@ -0,0 +1,58 @@
+
+import seedu.address.model.Model;
+
+
```
Missing javadoc description

yanbingtao replied:
Added in the latest commit. thanks!

ureshiiYing reviewed (commented):
Other comments: Perhaps naming the commands as `TodayCommand` and `TomorrowCommand` isn't the best? It felt vague to me. The missing javadoc descriptions did not help too.

yanbingtao committed:
Improve code quality.

yanbingtao commented:
> Other comments: Perhaps naming the commands as `TodayCommand` and `TomorrowCommand` isn't the best? It felt vague to me. The missing javadoc descriptions did not help too.

I've changed these two to `FindByTagTodayCommand.java` and `FindByTagTomorrowCommand.java`. And two Javadoc were also added for these two classes to better explain the meaning. Thanks!

Diwu-Yi reviewed (dismissed):
LGTM

yanbingtao committed:
Merge branch 'master' into branch-v1.3-c-tag-find-bt

Diwu-Yi reviewed (approved):
LGTM

Persdre reviewed (approved):
LGTM
